presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorPartitioningMetadata.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+import java.util.List;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorPartitioningMetadata
+{
+    private final ConnectorPartitioningHandle partitioningHandle;
+    private final List<String> partitionColumns;
+
+    public ConnectorPartitioningMetadata(ConnectorPartitioningHandle partitioningHandle, List<String> partitionColumns)
+    {
+        this.partitioningHandle = requireNonNull(partitioningHandle, "partitioningHandle is null");
+        this.partitionColumns = unmodifiableList(requireNonNull(partitionColumns, "partitionColumns is null"));
```

Comment:
Why not `ImmutableList.copyOf`?

Reply:
Unfortunately there's no guava in the SPI module =\